src/engine/bufferscalers/rubberbandwrapper.cpp

```diff
@@ -0,0 +1,172 @@
+#include "engine/bufferscalers/rubberbandwrapper.h"
+
+#include "engine/bufferscalers/rubberbandworker.h"
+#include "engine/bufferscalers/rubberbandworkerpool.h"
+#include "engine/engine.h"
+#include "util/assert.h"
+
+using RubberBand::RubberBandStretcher;
+
+#define RUBBERBANDV3 (RUBBERBAND_API_MAJOR_VERSION >= 2 && RUBBERBAND_API_MINOR_VERSION >= 7)
+
+namespace {
+
+mixxx::audio::ChannelCount getChannelPerWorker() {
+    RubberBandWorkerPool* pPool = RubberBandWorkerPool::instance();
+    return pPool ? pPool->channelPerWorker() : mixxx::kEngineChannelCount;
+}
+} // namespace
+
+int RubberBandWrapper::getEngineVersion() const {
+    VERIFY_OR_DEBUG_ASSERT(isValid()) {
+        return -1;
+    }
+#if RUBBERBANDV3
+    return m_pInstances[0]->getEngineVersion();
+#else
+    return 2;
+#endif
+}
+void RubberBandWrapper::setTimeRatio(double ratio) {
+    for (auto& stretcher : m_pInstances) {
+        stretcher->setTimeRatio(ratio);
+    }
+}
+size_t RubberBandWrapper::getSamplesRequired() const {
+    size_t require = 0;
+    for (auto& stretcher : m_pInstances) {
+        require = qMax(require, stretcher->getSamplesRequired());
+    }
+    return require;
+}
+int RubberBandWrapper::available() const {
+    int available = std::numeric_limits<int>::max();
+    for (auto& stretcher : m_pInstances) {
+        available = qMin(available, stretcher->available());
+    }
+    return available;
+}
```

Comment:
this will return int max when the vector is empty. That doesn't seem right. Should be 0 imo.